File: iteration3/Test/RegistrationTest.py
import unittest

from iteration3.Advisor import Advisor
from iteration3.Course import Course
from iteration3.Registration import Registration
from iteration3.Student import Student
from iteration3.Transcript import Transcript


class TestRegistration(unittest.TestCase):
    def setUp(self):
        self.student = Student(150120000, "name", "surname", "username", "password", 3)
        self.course1 = Course("courseName1", "CSE100", "Mandatory", 3, 3, 120, 8, "Monday")
        self.course2 = Course("courseName2", "CSE101", "Mandatory", 4, 3, 120, 9, "Monday")
        self.courses = [self.course1, self.course2]
        self.registration = Registration(self.student, self.courses)

    def test_rejectRequest(self):
        self.registration.rejectRequest()
        self.assertIs(self.student.isHasRequest(), False)
        self.assertEqual(len(self.student.getDraft()), 0)

    def test_approveRequest(self):
        transcript = Transcript(self.student)
        self.student.setTranscript(transcript)
        self.registration.approveRequest()
        self.assertEqual(len(self.student.getTranscript().getStudentCourses()), 2)
        self.assertIs(self.student.isHasRequest(), False)
        self.assertEqual(len(self.student.getDraft()), 0)

    def test_addRequest(self):
        advisor = Advisor(1234567, "name", "surname", "username", "password")
        self.registration.addRequest(advisor)
        self.assertIn(self.registration, advisor.getRequestList())
        self.assertIs(self.student.isHasRequest(), True)


if __name__ == "__main__":
    unittest.main()

File: iteration1/Test/RegistrationTest.java
package iteration1.Test;
import iteration1.*;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import iteration1.Course;
import iteration1.Student;
import iteration1.Transcript;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;


public class RegistrationTest {
    private List<Course> courses;
    private Course course1;
    private Course course2;
    private Registration registration;

    private Student student;

    @BeforeEach
    void setUp() {
        student = new Student(150120000, "name", "surname", "username", "password", (byte) 3);
        course1 = new Course("courseName1", "CSE100", 3, (byte) 3);
        course2 = new Course("courseName2", "CSE101", 4, (byte) 3);
        courses = List.of(course1, course2);
        registration = new Registration(student, courses);
    }

    @Test

    void rejectRequestTest() {
        registration.rejectRequest();
        assertFalse(student.isHasRequest());
        assertEquals(0, student.getDraft().size());
    }
    @Test
    void approveRequestTest() {
        Transcript transcript = new Transcript(student);
        student.setTranscript(transcript);
        registration.approveRequest();

        assertEquals(2, student.getTranscript().getStudentCourses().size());
        assertFalse(student.isHasRequest());
        assertEquals(0, student.getDraft().size());
    }
    
    @Test
    void addRequestTest() {
        Advisor advisor = new Advisor(1234567,"name" ,"surname","username","password");
        registration.addRequest(advisor);
        assertTrue(advisor.getRequests().contains(registration));
        assertTrue(student.isHasRequest());
    }
}


